tsp-output/@azure-tools/typespec-csharp/src/Generated/Audio.cs

```diff
@@ -123,60 +115,59 @@ public virtual async Task<Result> CreateSpeechAsync(RequestBody content, Request
-        /// <exception cref="MessageFailedException"> Service returned a non-success status code. </exception>
+        /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
         /// <returns> The response returned from the service. </returns>
-        public virtual Result CreateSpeech(RequestBody content, RequestOptions context = null)
+        public virtual ClientResult CreateSpeech(BinaryContent content, RequestOptions options = null)
         {
-            ClientUtilities.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNull(content, nameof(content));
 
-            using var scope = ClientDiagnostics.CreateSpan("Audio.CreateSpeech");
-            scope.Start();
-            try
-            {
-                using PipelineMessage message = CreateCreateSpeechRequest(content, context);
-                return Result.FromResponse(_pipeline.ProcessMessage(message, context));
-            }
-            catch (Exception e)
+            options ??= new RequestOptions();
+
+            using PipelineMessage message = CreateCreateSpeechRequest(content, options);
+
+            _pipeline.Send(message);
+
+            PipelineResponse response = message.Response!;
+
+            if (response.IsError && options.ErrorOptions == ClientErrorBehaviors.Default)
             {
-                scope.Failed(e);
-                throw;
+                throw new ClientResultException(response);
             }
+
+            return ClientResult.FromResponse(response);
         }
 
         /// <summary> Transcribes audio into the input language. </summary>
         /// <param name="audio"> The <see cref="CreateTranscriptionRequest"/> to use. </param>
-        /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="audio"/> is null. </exception>
-        public virtual async Task<Result<CreateTranscriptionResponse>> CreateTranscriptionAsync(CreateTranscriptionRequest audio, CancellationToken cancellationToken = default)
+        public virtual async Task<ClientResult<CreateTranscriptionResponse>> CreateTranscriptionAsync(CreateTranscriptionRequest audio)
         {
-            ClientUtilities.AssertNotNull(audio, nameof(audio));
+            Argument.AssertNotNull(audio, nameof(audio));
 
-            RequestOptions context = FromCancellationToken(cancellationToken);
-            using RequestBody content = audio.ToRequestBody();
-            Result result = await CreateTranscriptionAsync(content, context).ConfigureAwait(false);
-            return Result.FromValue(CreateTranscriptionResponse.FromResponse(result.GetRawResponse()), result.GetRawResponse());
+            (BinaryContent content, string contentType, RequestOptions options) = await audio.ToMultipartContentAsync().ConfigureAwait(false);
```

Comment:
This is where we obtain the multipart content instead of standard BinaryContent.

Reply:
This part looks different in https://github.com/annelo-msft/openai-in-typespec/pull/2 and https://github.com/annelo-msft/openai-in-typespec/pull/3.

Reply:
https://github.com/Azure/autorest.csharp/issues/4340